File: boj/17626.go
package main

import (
	"fmt"
	"math"
)

func nearSquare(num int) int {
	n := int(math.Round(math.Sqrt(float64(num))))
	for n*n < num {
		n++
	}
	if n*n > num {
		n--
	}
	fmt.Printf("input:%d, n:%d, result:%d\n", num, n, n*n)
	return n * n
}

func main() {
	var num int
	fmt.Scanf("%d", &num)
	cnt := 0

	for num > 0 {
		num -= nearSquare(num)
		fmt.Println(num)
		cnt++
	}
	fmt.Println(cnt)
}

File: boj/17626.py
import math
def find(num, dp):
    r = 0
    i = 1
    while i**i <= num:
        if dp[i] == 0:
            dp[i] = i**i
            r = i
        i+=1
    print(r)
    return r

num = int(input())
print(num)


dp = [0] * 50000 
cnt = 0

answer = []
'''
while num > 0:
    t = math.floor(math.sqrt(num) + 1) 
    print("num:%d, t:%d" % (num,t))
    r = find(t, dp)  
    num -= r
    answer.append(r)
    print("r:%d" %(r))
'''

while num > 0:
    t = int(math.sqrt(num))
    print(t*t)
    
    if t * t > num:
        t-=1
    answer.append(t)
    num -= t*t
print(answer)
